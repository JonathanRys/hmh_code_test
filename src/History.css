.history {
  display: inline-block;
  padding: 10px;
  vertical-align: top;
}

s