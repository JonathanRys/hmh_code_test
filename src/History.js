import React, { Component } from 'react'
import PropTypes from 'prop-types'

class History extends Component {
    render() {
        return (
            <div className="history">
                <h3>History</h3>
                { this.props.history.map((item, key) => <div key={ `item-${key}` }>{ JSON.stringify(item) }</div>) }
            </div>
        )
    }
}

History.propTypes = {
    history: PropTypes.array.isRequired
}

export default History
