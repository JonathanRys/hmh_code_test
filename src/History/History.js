import React, { Component } from 'react'
import PropTypes from 'prop-types'

import './History.css'

class History extends Component {
    render() {
        return (
            <div className="history">
                <h3>History</h3>
                {this.props.history.map((item, key) => <div key={`item-${key}`}>{JSON.stringify(item)}</div>)}
            </div>
        )
    }
}

History.propTypes = {
    history: PropTypes.array.isRequired
}

export default History
