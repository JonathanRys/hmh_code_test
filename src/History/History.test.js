
import React from 'react';
import ReactDOM from 'react-dom';
import History from './History';


describe('Test History component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<History history={[]} />, div);
    });

    it('renders without crashing', () => {
        const div = document.createElement('div');
        expect(ReactDOM.render(<History history={[{}, {
            firstName: "Lester",
            lastName: "Tester",
            startDate: "Today"
        }]} />, div)).toMatchSnapshot()
    });
})
