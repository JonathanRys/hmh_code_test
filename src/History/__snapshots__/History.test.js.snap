// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test History component renders without crashing 1`] = `
History {
  "_reactInternalFiber": FiberNode {
    "_debugID": 9,
    "_debugIsCurrentlyTiming": false,
    "_debugOwner": null,
    "_debugSource": null,
    "alternate": null,
    "child": FiberNode {
      "_debugID": 10,
      "_debugIsCurrentlyTiming": false,
      "_debugOwner": [Circular],
      "_debugSource": null,
      "alternate": null,
      "child": FiberNode {
        "_debugID": 11,
        "_debugIsCurrentlyTiming": false,
        "_debugOwner": [Circular],
        "_debugSource": null,
        "alternate": null,
        "child": null,
        "effectTag": 0,
        "expirationTime": 0,
        "firstEffect": null,
        "index": 0,
        "internalContextTag": 0,
        "key": null,
        "lastEffect": null,
        "memoizedProps": Object {
          "children": "History",
        },
        "memoizedState": null,
        "nextEffect": null,
        "pendingProps": null,
        "ref": null,
        "return": [Circular],
        "sibling": FiberNode {
          "_debugID": 12,
          "_debugIsCurrentlyTiming": false,
          "_debugOwner": null,
          "_debugSource": null,
          "alternate": null,
          "child": FiberNode {
            "_debugID": 13,
            "_debugIsCurrentlyTiming": false,
            "_debugOwner": [Circular],
            "_debugSource": null,
            "alternate": null,
            "child": null,
            "effectTag": 0,
            "expirationTime": 0,
            "firstEffect": null,
            "index": 0,
            "internalContextTag": 0,
            "key": "item-0",
            "lastEffect": null,
            "memoizedProps": Object {
              "children": "{}",
            },
            "memoizedState": null,
            "nextEffect": null,
            "pendingProps": null,
            "ref": null,
            "return": [Circular],
            "sibling": FiberNode {
              "_debugID": 14,
              "_debugIsCurrentlyTiming": false,
              "_debugOwner": [Circular],
              "_debugSource": null,
              "alternate": null,
              "child": null,
              "effectTag": 0,
              "expirationTime": 0,
              "firstEffect": null,
              "index": 1,
              "internalContextTag": 0,
              "key": "item-1",
              "lastEffect": null,
              "memoizedProps": Object {
                "children": "{\\"firstName\\":\\"Lester\\",\\"lastName\\":\\"Tester\\",\\"startDate\\":\\"Today\\"}",
              },
              "memoizedState": null,
              "nextEffect": null,
              "pendingProps": null,
              "ref": null,
              "return": [Circular],
              "sibling": null,
              "stateNode": <div>
                {"firstName":"Lester","lastName":"Tester","startDate":"Today"}
              </div>,
              "tag": 5,
              "type": "div",
              "updateQueue": null,
            },
            "stateNode": <div>
              {}
            </div>,
            "tag": 5,
            "type": "div",
            "updateQueue": null,
          },
          "effectTag": 0,
          "expirationTime": 0,
          "firstEffect": null,
          "index": 1,
          "internalContextTag": 0,
          "key": null,
          "lastEffect": null,
          "memoizedProps": Array [
            <div>
              {}
            </div>,
            <div>
              {"firstName":"Lester","lastName":"Tester","startDate":"Today"}
            </div>,
          ],
          "memoizedState": null,
          "nextEffect": null,
          "pendingProps": null,
          "ref": null,
          "return": [Circular],
          "sibling": null,
          "stateNode": null,
          "tag": 10,
          "type": null,
          "updateQueue": null,
        },
        "stateNode": <h3>
          History
        </h3>,
        "tag": 5,
        "type": "h3",
        "updateQueue": null,
      },
      "effectTag": 0,
      "expirationTime": 0,
      "firstEffect": null,
      "index": 0,
      "internalContextTag": 0,
      "key": null,
      "lastEffect": null,
      "memoizedProps": Object {
        "children": Array [
          <h3>
            History
          </h3>,
          Array [
            <div>
              {}
            </div>,
            <div>
              {"firstName":"Lester","lastName":"Tester","startDate":"Today"}
            </div>,
          ],
        ],
        "className": "history",
      },
      "memoizedState": null,
      "nextEffect": null,
      "pendingProps": null,
      "ref": null,
      "return": [Circular],
      "sibling": null,
      "stateNode": <div
        class="history"
      >
        <h3>
          History
        </h3>
        <div>
          {}
        </div>
        <div>
          {"firstName":"Lester","lastName":"Tester","startDate":"Today"}
        </div>
      </div>,
      "tag": 5,
      "type": "div",
      "updateQueue": null,
    },
    "effectTag": 1,
    "expirationTime": 0,
    "firstEffect": null,
    "index": 0,
    "internalContextTag": 0,
    "key": null,
    "lastEffect": null,
    "memoizedProps": Object {
      "history": Array [
        Object {},
        Object {
          "firstName": "Lester",
          "lastName": "Tester",
          "startDate": "Today",
        },
      ],
    },
    "memoizedState": null,
    "nextEffect": null,
    "pendingProps": null,
    "ref": null,
    "return": FiberNode {
      "_debugID": 7,
      "_debugIsCurrentlyTiming": false,
      "_debugOwner": null,
      "_debugSource": null,
      "alternate": FiberNode {
        "_debugID": 7,
        "_debugIsCurrentlyTiming": false,
        "_debugOwner": null,
        "_debugSource": null,
        "alternate": [Circular],
        "child": null,
        "effectTag": 0,
        "expirationTime": 1,
        "firstEffect": null,
        "index": 0,
        "internalContextTag": 0,
        "key": null,
        "lastEffect": null,
        "memoizedProps": null,
        "memoizedState": null,
        "nextEffect": null,
        "pendingProps": null,
        "ref": null,
        "return": null,
        "sibling": null,
        "stateNode": Object {
          "containerInfo": <div>
            <div
              class="history"
            >
              <h3>
                History
              </h3>
              <div>
                {}
              </div>
              <div>
                {"firstName":"Lester","lastName":"Tester","startDate":"Today"}
              </div>
            </div>
          </div>,
          "context": Object {},
          "current": [Circular],
          "finishedWork": null,
          "hydrate": false,
          "isReadyForCommit": false,
          "nextScheduledRoot": null,
          "pendingChildren": null,
          "pendingContext": null,
          "remainingExpirationTime": 0,
        },
        "tag": 3,
        "type": null,
        "updateQueue": Object {
          "baseState": null,
          "callbackList": null,
          "expirationTime": 1,
          "first": Object {
            "callback": null,
            "expirationTime": 1,
            "isForced": false,
            "isReplace": false,
            "next": null,
            "nextCallback": null,
            "partialState": Object {
              "element": <History
                history={
                  Array [
                    Object {},
                    Object {
                      "firstName": "Lester",
                      "lastName": "Tester",
                      "startDate": "Today",
                    },
                  ]
                }
              />,
            },
          },
          "hasForceUpdate": false,
          "isInitialized": false,
          "isProcessing": false,
          "last": Object {
            "callback": null,
            "expirationTime": 1,
            "isForced": false,
            "isReplace": false,
            "next": null,
            "nextCallback": null,
            "partialState": Object {
              "element": <History
                history={
                  Array [
                    Object {},
                    Object {
                      "firstName": "Lester",
                      "lastName": "Tester",
                      "startDate": "Today",
                    },
                  ]
                }
              />,
            },
          },
        },
      },
      "child": [Circular],
      "effectTag": 0,
      "expirationTime": 0,
      "firstEffect": [Circular],
      "index": 0,
      "internalContextTag": 0,
      "key": null,
      "lastEffect": [Circular],
      "memoizedProps": null,
      "memoizedState": Object {
        "element": <History
          history={
            Array [
              Object {},
              Object {
                "firstName": "Lester",
                "lastName": "Tester",
                "startDate": "Today",
              },
            ]
          }
        />,
      },
      "nextEffect": null,
      "pendingProps": null,
      "ref": null,
      "return": null,
      "sibling": null,
      "stateNode": Object {
        "containerInfo": <div>
          <div
            class="history"
          >
            <h3>
              History
            </h3>
            <div>
              {}
            </div>
            <div>
              {"firstName":"Lester","lastName":"Tester","startDate":"Today"}
            </div>
          </div>
        </div>,
        "context": Object {},
        "current": [Circular],
        "finishedWork": null,
        "hydrate": false,
        "isReadyForCommit": false,
        "nextScheduledRoot": null,
        "pendingChildren": null,
        "pendingContext": null,
        "remainingExpirationTime": 0,
      },
      "tag": 3,
      "type": null,
      "updateQueue": null,
    },
    "sibling": null,
    "stateNode": [Circular],
    "tag": 2,
    "type": [Function],
    "updateQueue": null,
  },
  "_reactInternalInstance": Object {},
  "context": Object {},
  "props": Object {
    "history": Array [
      Object {},
      Object {
        "firstName": "Lester",
        "lastName": "Tester",
        "startDate": "Today",
      },
    ],
  },
  "refs": Object {},
  "state": null,
  "updater": Object {
    "enqueueForceUpdate": [Function],
    "enqueueReplaceState": [Function],
    "enqueueSetState": [Function],
    "isMounted": [Function],
  },
}
`;
